### PR TITLE
add gcs rehydration instructions and remove beta flag for rehydration

### PR DESCRIPTION
**Pull request by estib**
### What does this PR do?
1. Adds instructions for GCS setup in the Logs Rehydration doc.
2. Removes the beta flag for rehydration, (also removes the reference to supporting US region only) 

### Motivation

Rehydration is available in GCS now
Rehydration is available for all customers, including the EU, once this PR is merged: https://github.com/DataDog/consul-config/pull/41843/files

### Preview link
https://docs-staging.datadoghq.com/estib/rehydration-gcs/logs/archives/rehydrating?tab=googlecloudstorage

### Additional Notes
We can merge this docs pr when this PR is merged: https://github.com/DataDog/consul-config/pull/41843/files

### CODE REVIEW

File: content/en/logs/archives/rehydrating.md [hunk 1]
```diff
@@ -119,8 +129,5 @@ Datadog would love to receive your feedback on this feature, especially as it is
 [4]: #permissions
 [5]: /logs/explorer/search
 [6]: /logs/?tab=ussite#reserved-attributes
-[7]: /logs/archives/s3/?tab=roledelegation#create-and-configure-an-s3-bucket
-[8]: https://app.datadoghq.com/account/settings#integrations/amazon-web-services
-[9]: /integrations/amazon_web_services/?tab=allpermissions#installation
-[10]: https://app.datadoghq.com/logs/pipelines/archives
-[11]: https://forms.gle/X4jhi13Rd2pFSuSHA
+[7]: /logs/archives/
+[8]: https://forms.gle/X4jhi13Rd2pFSuSHA
```
Comment (ruthnaebeck) [inline content/en/logs/archives/rehydrating.md]: This link isn't used

File: content/en/logs/archives/rehydrating.md [hunk 2]
```diff
@@ -103,13 +100,26 @@ In order to rehydrate log events from your archives, Datadog uses the IAM Role i
...
-### Feedback
+[1]: https://app.datadoghq.com/account/settings#integrations/amazon-web-services
+[2]: /integrations/amazon_web_services/?tab=allpermissions#installation
+[3]: https://app.datadoghq.com/logs/pipelines/archives
+
+{{% /tab %}}
+
+{{% tab "Google Cloud Storage" %}}
+
+In order to rehydrate log events from your archives, Datadog uses a Service Account with the Storage Object Viewer role. You can grant this role to your Datadog Service Account from the [GCP IAM Admin page][1] by editing the Service Account's permissions, adding another role, and then selecting Storage > Storage Object Viewer.
```
Comment (ruthnaebeck) [inline content/en/logs/archives/rehydrating.md]:
```suggestion
In order to rehydrate log events from your archives, Datadog uses a service account with the Storage Object Viewer role. You can grant this role to your Datadog service account from the [GCP IAM Admin page][1] by editing the service account's permissions, adding another role, and then selecting Storage > Storage Object Viewer. 
```

Comment (estib) [reply content/en/logs/archives/rehydrating.md]: cool -- updated this in my latest commit, thanks!

File: content/en/logs/archives/rehydrating.md (hunk 1 above)
Comment (estib) [reply content/en/logs/archives/rehydrating.md]: great catch!

**Pull request merged**